### PR TITLE
chore: improve logging

### PR DESCRIPTION
**Pull request by abc3**

### CODE REVIEW

File: lib/supavisor/client_handler.ex [hunk 1]
```diff
@@ -71,37 +71,39 @@ defmodule Supavisor.ClientHandler do
 
   @impl true
   def handle_event(:info, {_proto, _, <<"GET", _::binary>>}, :exchange, data) do
-    Logger.debug("Client is trying to request HTTP")
+    Logger.debug("ClientHanlder: Client is trying to request HTTP")
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:74]:
```suggestion
    Logger.debug("ClientHandler: Client is trying to request HTTP")
```

Also everywhere else.

Comment (chasers) [reply lib/supavisor/client_handler.ex:74]: Could also do __MODULE__

Comment (abc3) [reply lib/supavisor/client_handler.ex:74]: > Hanlder
🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦

File: lib/supavisor/client_handler.ex [hunk 2]
```diff
@@ -327,34 +338,36 @@ defmodule Supavisor.ClientHandler do
...
     HH.sock_send(data.sock, Server.application_name())
     {:keep_state_and_data, {:timeout, data.heartbeat_interval, :heartbeat_check}}
   end
 
   # handle Terminate message
   def handle_event(:info, {proto, _, <<?X, 4::32>>}, :idle, _)
       when proto in [:tcp, :ssl] do
-    Logger.debug("Receive termination")
-    {:stop, :normal}
+    Logger.debug("ClientHanlder: Receive termination")
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:363]: This means the client close the connection yeah? 

Can we say something like "Received connection close message"

Comment (abc3) [reply lib/supavisor/client_handler.ex:363]: A client sent the termination query, and then we stopped the handler

File: lib/supavisor/client_handler.ex [hunk 3]
```diff
@@ -404,79 +417,74 @@ defmodule Supavisor.ClientHandler do
...
-  def handle_event(_, {:tcp_closed, _}, _, data) do
-    Logger.debug("tcp soket closed for #{inspect(data.tenant)}")
-    {:stop, :normal}
-  end
-
   def handle_event(_, {closed, _}, _, data)
       when closed in [:tcp_closed, :ssl_closed] do
-    Logger.debug("#{closed} soket closed for #{inspect(data.tenant)}")
-    {:stop, :normal}
+    Logger.debug("ClientHanlder: #{closed} soket closed for #{inspect(data.tenant)}")
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:456]:
```suggestion
    Logger.debug("ClientHanlder: #{closed} socket closed for #{inspect(data.tenant)}")
```

File: lib/supavisor/client_handler.ex [hunk 4]
```diff
@@ -404,79 +417,74 @@ defmodule Supavisor.ClientHandler do
...
+    {:stop, {:shutdown, :socket_closed}}
   end
 
   # linked DbHandler went down
   def handle_event(:info, {:EXIT, db_pid, reason}, _, _) do
-    Logger.error("DB handler #{inspect(db_pid)} exited #{inspect(reason)}")
-    {:stop, :normal}
+    Logger.error("ClientHanlder: DB handler #{inspect(db_pid)} exited #{inspect(reason)}")
```
Comment (chasers) [inline lib/supavisor/client_handler.ex:462]:
```suggestion
    Logger.error("ClientHanlder: DbHandler #{inspect(db_pid)} exited #{inspect(reason)}")
```

**Review by chasers — APPROVED** [review]

File: lib/supavisor/client_handler.ex [hunk 5]
```diff
@@ -534,24 +542,29 @@ defmodule Supavisor.ClientHandler do
...
+    case Db.get_state(pid) do
+      {:ok, :busy} ->
+        Logger.warning(
+          "ClientHanlder: Kill DbHandler #{inspect(pid)} and termination, reason #{inspect(reason)}"
+        )
+
+        Db.stop(pid)
+
+      _ ->
+        Logger.warning("ClientHanlder: termination with reason #{inspect(reason)}")
```
[the remark this answers is not in the document]
Comment (abc3) [reply lib/supavisor/client_handler.ex:560]: this is exactly the termination function and a reason will be logged as well

File: lib/supavisor/client_handler.ex (hunk 2 above)
Comment (chasers) [reply lib/supavisor/client_handler.ex]: The point is someone should be able to read it and know "client is asking us to close this connection"

"Receive termination" doesn't really express that.

File: lib/supavisor/client_handler.ex [hunk 6]
```diff
@@ -404,79 +417,74 @@ defmodule Supavisor.ClientHandler do
...
         else
-          Logger.debug("DbHandler call buffering")
+          Logger.debug("ClientHandler: DbHandler call buffering")
           :keep_state_and_data
         end
 
       {:error, reason} ->
         msg = "DbHandler error: #{inspect(reason)}"
-        Logger.error(msg)
+        Logger.error("ClientHandler: " <> msg)
```
Comment (abc3) [inline lib/supavisor/client_handler.ex:438]:
```suggestion
        Logger.error("ClientHandler: #{msg}")
```

File: lib/supavisor/client_handler.ex [hunk 7]
```diff
@@ -404,79 +417,74 @@ defmodule Supavisor.ClientHandler do
...
         :keep_state_and_data
 
       {:buffering, size} ->
-        Logger.debug("DbHandler call buffering #{size}")
+        Logger.debug("ClientHandler: DbHandler call buffering #{size}")
 
         if size > 1_000_000 do
           msg = "DbHandler buffer size is too big: #{size}"
-          Logger.error(msg)
+          Logger.error("ClientHandler: " <> msg)
```
Comment (abc3) [inline lib/supavisor/client_handler.ex:428]:
```suggestion
          Logger.error("ClientHandler: #{msg}")
```

File: lib/supavisor/db_handler.ex [hunk 8]
```diff
@@ -384,13 +396,15 @@ defmodule Supavisor.DbHandler do
   # linked client_handler went down
   def handle_event(_, {:EXIT, pid, reason}, state, data) do
     if reason != :normal do
-      Logger.error("Client handler #{inspect(pid)} went down with reason #{inspect(reason)}")
+      Logger.error(
+        "DbHandler: Client handler #{inspect(pid)} went down with reason #{inspect(reason)}"
```
Comment (abc3) [inline lib/supavisor/db_handler.ex:400]:
```suggestion
        "DbHandler: Clienthandler #{inspect(pid)} went down with reason #{inspect(reason)}"
```

File: lib/supavisor/db_handler.ex [hunk 9]
```diff
@@ -384,13 +396,15 @@ defmodule Supavisor.DbHandler do
   # linked client_handler went down
   def handle_event(_, {:EXIT, pid, reason}, state, data) do
     if reason != :normal do
-      Logger.error("Client handler #{inspect(pid)} went down with reason #{inspect(reason)}")
+      Logger.error(
+        "DbHandler: Clienthandler #{inspect(pid)} went down with reason #{inspect(reason)}"
```
Comment (abc3) [inline lib/supavisor/db_handler.ex:400]:
```suggestion
        "DbHandler: ClientHandler #{inspect(pid)} went down with reason #{inspect(reason)}"
```

**Pull request merged**